.github/workflows/test-upgrade.yaml

```
@@ -133,3 +132,180 @@ jobs:
         run: |
           kubectl wait --timeout=300s --for=condition=Ready kuadrant kuadrant -n ${{ inputs.kuadrantNamespace }}
           echo "kuadrant upgrade ✅"
+  olm-upgrade-test:
+    runs-on: ubuntu-latest
+    name: OLM Upgrade Test
+    if: ${{ github.event_name == 'workflow_dispatch' }}
+    env:
+      KIND_CLUSTER_NAME: kuadrant-test
+      K8S_USER: kuadrant-admin  # can be whatever, it does not matter.
+      CLUSTER_NAME: remote-cluster
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+      - name: Install helm
+        run: |
+          make helm
+      - name: Install yq tool
+        run: |
+          # following sub-shells running make target should have yq already installed
+          make yq
+      - name: Deploy local Kind cluster
+        if: ${{ github.event_name == 'workflow_dispatch' && inputs.clusterServer == '' }}
```

as in the helm version, we could do the job scoped env:

```yaml
env:
  LOCAL_CLUSTER: ${{ inputs.clusterServer == '' && inputs.clusterToken != ''}}
```

thus in checks like this, we could do:

```suggestion
        if: ${{ !env.LOCAL_CLUSTER }}
```

The `github.event_name == 'workflow_dispatch'` wouldn't be needed neither, wdyt?

---

ok, doing like that, lgtm